packages/gitbook-docs/src/markdoc/schema/pageLink.ts

```diff
@@ -0,0 +1,23 @@
+import { Schema, Tag } from '@markdoc/markdoc';
+import { LinkAttr } from '../attributes';
+
+export type PageLinkProps<A> = {
+  readonly url: string;
+  readonly children: A;
+};
+
+export const pageLink: Schema = {
+  children: ['Link'],
+  attributes: {
+    url: {
+      type: LinkAttr,
+      required: true,
+    },
+  },
+  transform: (node, config) => {
+    const children = Array.from(node.walk())
```

Comment:
```suggestion
    // To avoid to render a paragraph as child of PageLink, we Search the `link` types within the children and transform them
    const children = Array.from(node.walk())
```